content/Registration and Authentication in your Django app with dj-rest-auth- final - Bruno Michetti.md

```diff
@@ -0,0 +1,497 @@
+
+```python
+# Your settings file
+...
+CUSTOM_PASSWORD_RESET_CONFIRM = 'desired URL'
+```
+
+Then you have to create two files under `myapp/templatetags/`: One has to be an empty `__init__.py` file, and the other one registers and gets the tags. Let's call it `password_reset_template_load.py`:
+
+```python
+from django import template
+from django.conf import settings
+
+register = template.Library()
+
+
+@register.simple_tag
+def get_settings_var(name):
+    return getattr(settings, name)
+```
+
+Finally, go to your customized email template that you have created to overwrite the default one, and load the defined value by adding `{% load password_reset_template_load %}` at the top of the file, and after that add the line that has the custom value (in this particular case don't forget the `uid` and `token`):
+
+```html
+// Your template {% load password_reset_template_load %} ... {% get_settings_var
+'CUSTOM_PASSWORD_RESET_CONFIRM' %}?uidb64={{ uid }}&token={{ token }}
+```
+
+With these settings you have defined your custom template tags. I'm going to show you the final project structure with all the customization that I talked about:
+
+```
+root-project-folder
+    |__myapp
+    |    |__templates
+    |    |   |__account
+    |    |   |   |__email
+    |    |   |        |__email_confirmation_message.txt
+    |    |   |__registration
+    |    |       |__password_reset_email.html
+    |    |__templatetags
+    |        |__password_reset_template_load.py
+    ...
+```
+
+From now on, the email for password reset will show a customized URL. Remember that this was an example, you can create any template tag you want and use it in any template you have.
+
+#### Notes
+
+1. If while trying this you have an error that shows `Site matching query does not exist.`, you can solve it by adding `SITE_ID = 1` to your settings.
+2. When you create your own templatetags make sure that the main app is added in the `INSTALLED_APPS` list in your settings.
+
+### Summary
+
+I have presented a very useful package to handle registration and authentication with a set of REST API endpoints.
+With minimal configuration, you can add endpoints to your Django app that already have those features implemented. We reviewed some advantages of the package, how easy it is to customize it, and code snippet examples.  
+Besides, I have shown you how to implement email validation at sign-up, and how to customize the emails that are sent by your app. We learned about the reset password functionality, and finally how to customize the templates with your own desired values. I recommend using the presented package because it solves lots of common problems and provides a set of features that are very important for the most of the existent applications. Maybe in the future, you can not only use it but also contribute to it.
```

Comment:
```suggestion
In addition to this, I have shown you how to implement email validation at sign-up, and how to customize the emails that are sent by your app. We learned about the reset password functionality, and finally how to customize the templates with your own desired values. I recommend using the presented package because it solves lots of common problems and provides a set of features that are very important for most applications. Maybe in the future, you can not only use it but also contribute to it.
```